price_service/server/src/vaa.ts

```diff
@@ -0,0 +1,73 @@
+import { ParsedVaa } from "@certusone/wormhole-sdk";
+import { GuardianSet } from "@certusone/wormhole-spydk/lib/cjs/proto/publicrpc/v1/publicrpc";
+import { ethers } from "ethers";
+
+const WormholeClusters = ["localnet", "testnet", "mainnet"] as const;
+export type WormholeCluster = typeof WormholeClusters[number];
+
+export function wormholeClusterFromString(s: string): WormholeCluster {
+  if (WormholeClusters.includes(s as WormholeCluster)) {
+    return s as WormholeCluster;
+  }
+  throw new Error(`Invalid wormhole cluster: ${s}`);
+}
+
+const guardianSets: Record<WormholeCluster, GuardianSet> = {
+  localnet: {
+    index: 0,
+    addresses: ["0xbeFA429d57cD18b7F8A4d91A2da9AB4AF05d0FBe"],
+  },
+  testnet: {
+    index: 0,
+    addresses: ["0x13947Bd48b18E53fdAeEe77F3473391aC727C638"],
+  },
+  mainnet: {
+    index: 3,
+    addresses: [
+      "0x58CC3AE5C097b213cE3c81979e1B9f9570746AA5",
```

Comment:
yeah it will break. Hopefully by the next guardian set change we won't have this service running anymore 😅 But if we had this running we should let it accept both the old and the new. We can periodically check the source of truth for the guardian (the wormhole eth contract) but to have 0 downtime we still need to support transition.

Reply:
you should be able to point use the grpc call to get the current guardian set
something similar to how these work - https://github.com/wormhole-foundation/wormhole/tree/main/sdk/js/src/rpc

```
import { publicrpc } from "@certusone/wormhole-sdk-proto-web";
const { GrpcWebImpl, PublicRPCServiceClientImpl } = publicrpc;
import { NodeHttpTransport } from "@improbable-eng/grpc-web-node-http-transport";

// need to use the NodeHttpTransport when running this in a node environment
export async function getGuardianSet(
  host: string,
  extraGrpcOpts = {transport: NodeHttpTransport()}
) {
  const rpc = new GrpcWebImpl(host, extraGrpcOpts);
  const api = new PublicRPCServiceClientImpl(rpc);
  return await api.GetCurrentGuardianSet({});
}
```